### ISSUE TITLE
CMake Policy CMP0148: ROS2 iron

### ISSUE THREAD

**Issue by maksymilian-minenko** [ros2/python_cmake_module#16]
OS: Ubuntu 22.04
CMake version: 3.29.2
ros2 version: iron
Steps to reproduce: Compile ROS 2 from source on Ubuntu 22.04.
Desired action: Set the cmake policy

Issue is similar to:
https://github.com/ros2/python_cmake_module/issues/14

The only difference it the ros2 version. Fixed was provided for ros2 rooling:
https://github.com/ros2/python_cmake_module/pull/7

But the same issue exits on ros2 iron.

```
CMake Warning (dev) at /home/maksymilian/ros2_iron/install/rosidl_generator_py/share/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake:20 (find_package):
  Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
  are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
  the cmake_policy command to set the policy and suppress this warning.
  ```
  ```
  $ cmake --help-policy CMP0148
CMP0148
-------

.. versionadded:: 3.27

The ``FindPythonInterp`` and ``FindPythonLibs`` modules are removed.

These modules have been deprecated since CMake 3.12.
CMake 3.27 and above prefer to not provide the modules.
This policy provides compatibility for projects that have not been
ported away from them.

Projects using the ``FindPythonInterp`` and/or ``FindPythonLibs``
modules should be updated to use one of their replacements:

* ``FindPython3``
* ``FindPython2``
* ``FindPython``

The ``OLD`` behavior of this policy is for ``find_package(PythonInterp)``
and ``find_package(PythonLibs)`` to load the deprecated modules.  The ``NEW``
behavior is for uses of the modules to fail as if they do not exist.

This policy was introduced in CMake version 3.27.  CMake version
3.27.0-rc3 warns when the policy is not set and uses ``OLD`` behavior.
Use the ``cmake_policy()`` command to set it to ``OLD`` or ``NEW``
explicitly.

.. note::
  The ``OLD`` behavior of a policy is
  ``deprecated by definition``
  and may be removed in a future version of CMake.
  ```

**Comment by clalancette** [comment]
So, this is only a problem in Iron if you upgrade the CMake version beyond what is shipped in Ubuntu 22.04.  Using the default CMake version (3.22.1), this warning does not occur.

The fix is complicated enough that I'm not a huge fan of backporting it.  And Iron is only supported until November.

Given all of these facts, I'm going to say we should just live with this warning and not fix it.  My suggestion is to use the version of CMake installed with Ubuntu 22.04, where you will not see these warnings.

I'm going to go ahead and close this out, but feel free to reopen or keep commenting.

**Comment by lc-look** [comment]
This problem might be solved by adding this code to your CMalkeLists.txt:
```
if(POLICY CMP0148) 
cmake_policy(SET CMP0148 OLD)
endif()
```
